packages/@react-stately/dnd/src/useDraggableCollectionState.ts

```diff
@@ -98,11 +98,11 @@ export function useDraggableCollectionState(props: DraggableCollectionStateOptio
     preview,
     getAllowedDropOperations,
     startDrag(key, event) {
-      selectionManager.setFocused(false);
-      setDragging(true);
       let keys = getKeys(key);
       draggingKeys.current = keys;
       draggedKey.current = key;
+      selectionManager.setFocused(false);
+      setDragging(true);
```

Comment:
I'm surprised that this was an issue, what was broken that relied on the dragging/draggedKeys refs being updated before these these lines were set?

Reply:
Well really the other one further down was the problem with `isDragging` persisting after drop, because the refs were updated after state. I also changed this one just in case React decided to re-render synchronously for some reason and we accessed the dragging keys before they were set.